pkg/sdk/event_tables_def.go

```diff
@@ -0,0 +1,148 @@
+	OptionalSQL("MAX_DATA_EXTENSION_TIME_IN_DAYS").
+	OptionalSQL("CHANGE_TRACKING").
+	OptionalSQL("COMMENT")
+
+var eventTableDropRowAccessPolicy = g.NewQueryStruct("EventTableDropRowAccessPolicy").
+	SQL("ROW ACCESS POLICY").
+	Identifier("Name", g.KindOfT[SchemaObjectIdentifier](), g.IdentifierOptions())
+
+var eventTableClusteringAction = g.NewQueryStruct("EventTableClusteringAction").
+	PredefinedQueryStructField("ClusterBy", "*[]string", g.KeywordOptions().Parentheses().SQL("CLUSTER BY")).
+	OptionalSQL("SUSPEND RECLUSTER").
+	OptionalSQL("RESUME RECLUSTER").
+	OptionalSQL("DROP CLUSTERING KEY")
+
+var searchOptimization = g.NewQueryStruct("SearchOptimization").
+	SQL("SEARCH OPTIMIZATION").
+	PredefinedQueryStructField("On", "[]string", g.KeywordOptions().SQL("ON"))
+
+var eventTableSearchOptimizationAction = g.NewQueryStruct("EventTableSearchOptimizationAction").
+	OptionalQueryStructField(
+		"Add",
+		searchOptimization,
+		g.KeywordOptions().SQL("ADD"),
+	).
+	OptionalQueryStructField(
+		"Drop",
+		searchOptimization,
+		g.KeywordOptions().SQL("DROP"),
+	)
+
+var EventTablesDef = g.NewInterface(
+	"EventTables",
+	"EventTable",
+	g.KindOfT[SchemaObjectIdentifier](),
+).CreateOperation(
+	"https://docs.snowflake.com/en/sql-reference/sql/create-event-table",
+	g.NewQueryStruct("CreateEventTable").
+		Create().
+		OrReplace().
+		SQL("EVENT TABLE").
+		IfNotExists().
+		Name().
+		PredefinedQueryStructField("ClusterBy", "[]string", g.KeywordOptions().Parentheses().SQL("CLUSTER BY")).
+		OptionalNumberAssignment("DATA_RETENTION_TIME_IN_DAYS", g.ParameterOptions()).
+		OptionalNumberAssignment("MAX_DATA_EXTENSION_TIME_IN_DAYS", g.ParameterOptions()).
+		OptionalBooleanAssignment("CHANGE_TRACKING", g.ParameterOptions()).
+		OptionalTextAssignment("DEFAULT_DDL_COLLATION", g.ParameterOptions().SingleQuotes()).
+		OptionalSQL("COPY GRANTS").
+		OptionalTextAssignment("COMMENT", g.ParameterOptions().SingleQuotes()).
+		PredefinedQueryStructField("RowAccessPolicy", "*RowAccessPolicy", g.KeywordOptions()).
+		OptionalTags().WithValidation(g.ValidIdentifier, "name"),
+).ShowOperation(
+	"https://docs.snowflake.com/en/sql-reference/sql/show-event-tables",
+	g.DbStruct("eventTableRow").
+		Field("created_on", "string").
+		Field("name", "string").
+		Field("database_name", "string").
+		Field("schema_name", "string").
+		Field("owner", "string").
+		Field("comment", "string").
+		Field("owner_role_type", "string").
+		Field("change_tracking", "string"),
+	g.PlainStruct("EventTable").
+		Field("CreatedOn", "string").
+		Field("Name", "string").
+		Field("DatabaseName", "string").
+		Field("SchemaName", "string").
+		Field("Owner", "string").
+		Field("Comment", "string").
+		Field("OwnerRoleType", "string").
+		Field("ChangeTracking", "bool"),
+	g.NewQueryStruct("ShowFunctions").
+		Show().
+		SQL("EVENT TABLES").
+		OptionalLike().
+		OptionalIn().
+		OptionalTextAssignment("STARTS WITH", g.ParameterOptions().SingleQuotes().NoEquals()).
```

Comment:
There is a `OptionalStartsWith()` function

Reply:
fixed